Source/WebCore/page/Page.cpp

```diff
@@ -5309,17 +5309,25 @@ void Page::updateFixedContainerEdges(BoxSideSet sides)
     if (!mainFrame)
         return;
 
+    RefPtr document = mainFrame->document();
+    if (!document)
+        return;
+
     RefPtr frameView = mainFrame->view();
     if (!frameView)
         return;
 
-    auto [edges, elements] = frameView->fixedContainerEdges([frameView, sides] {
+    auto [edges, elements] = frameView->fixedContainerEdges([&] {
         auto sidesToSample = sides;
         auto scrollOffset = frameView->scrollOffset();
         auto minimumOffset = frameView->minimumScrollOffset();
         auto maximumOffset = frameView->maximumScrollOffset();
 
-        if (scrollOffset.y() < minimumOffset.y())
+        bool canSampleTopEdge = settings().topContentInsetBackgroundCanChangeAfterScrolling()
+            || !frameView->wasEverScrolledExplicitlyByUser()
+            || document->parsing();
```

Comment:
The top edge can _not_ be sampled if all of the following:
- `TopContentInsetBackgroundCanChangeAfterScrolling` is enabled
- Main frame hasn't been explicitly scrolled by the user
- Document load has finished

Would that not read like this instead?


```suggestion
        bool canSampleTopEdge = settings().topContentInsetBackgroundCanChangeAfterScrolling()
            || frameView->wasEverScrolledExplicitlyByUser()
            || document->parsing();
```

Reply:
Ah, good catch. So the code is correct, but the commit message is wrong — the second point should read:

> 2.  The main frame has been scrolled explicitly by the user.

(...instead of _never been scrolled_). Thanks for the review!

Reply:
Makes more sense!